src/lib/comms/include/sol-socket.h

```diff
@@ -0,0 +1,315 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <sol-network.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup Socket Socket
+ * @ingroup Comms
+ *
+ * @brief API to create an endpoint communication using sockets.
+ *
+ * @{
+ */
+
+struct sol_socket;
+
+struct sol_socket_default_options {
+#ifndef SOL_NO_API_VERSION
+#define SOL_SOCKET_DEFAULT_OPTIONS_API_VERSION (1)  /**< compile time API version to be checked during runtime */
+    /**
+     * must match #SOL_SOCKET_DEFAULT_OPTIONS_API_VERSION at runtime.
+     */
+    uint16_t api_version;
+#endif
+
+    /**
+     * @brief Register a function to be called when the socket has data to
+     * be read.
+     */
+    bool (*on_can_read)(void *data, struct sol_socket *s);
+
+    /**
+     * @brief Register a function to be called when the socket is ready to
+     * be written.
+     */
+    bool (*on_can_write)(void *data, struct sol_socket *s);
+
+    /**
+     * @brief User data data will be given in on_can_read() and
+     * on_can_write()
+     */
+    const void *data;
+
+    /**
+     * @brief The family that should be used when creating the socket.
+     * @see sol_network_family
+     */
+    enum sol_network_family family;
+
+    /**
+     * @brief If the socket's data should be encrypted or not.
+     */
+    bool secure;
+
+    /**
+     * @brief Allows multiple sockets to be bound to the same
+     * socket address. It is used by sol_socket_bind()
+     */
+    bool reuse_port;
+
+    /**
+     * @brief Allow reuse of local addresses. It is used by
+     * sol_socket_bind()
+     */
+    bool reuse_addr;
```

Comment:
do we really need both? I didn't follow all the discussion about the socket options on linux :-(